libraries/integration/Microsoft.Bot.Builder.ApplicationInsights.Core/ServiceCollectionExtensions.cs

```diff
@@ -38,10 +37,10 @@ public static IServiceCollection AddBotApplicationInsights(this IServiceCollecti
 
             // Enables Bot Telemetry to save user/session id's as the bot user id and session
             services.AddMemoryCache();
-            services.TryAddTransient<TelemetrySaveBodyASPMiddleware>();
-            services.TryAddSingleton<ITelemetryInitializer>(new OperationCorrelationTelemetryInitializer());
-            services.TryAddSingleton<ITelemetryInitializer, TelemetryBotIdInitializer>();
-            services.TryAddSingleton<IBotTelemetryClient, BotTelemetryClient>();
+            services.AddTransient<TelemetrySaveBodyASPMiddleware>();
+            services.AddSingleton<ITelemetryInitializer>(new OperationCorrelationTelemetryInitializer());
+            services.AddSingleton<ITelemetryInitializer, TelemetryBotIdInitializer>();
+            services.AddSingleton<IBotTelemetryClient, BotTelemetryClient>();
```

Comment:
If these all must be added why are they separate services?
In other words, if there was some scenario whereby you could plug in an alternative subcomponent then I could imagine why you would want to keep them separate otherwise I don't understand why you wouldn't just make them all the same object.

Reply:
Maybe can split the difference here - we can keep it in, but then provide a sample overriding BotTelemetryClient (I know Gabo is going to do this).